src/tasks/WasmAppBuilder/ReadEmccVersion.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using Microsoft.Build.Utilities;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.WebAssembly.Build.Tasks
+{
+    public class ReadEmccVersion : Task
+    {
+        [NotNull]
+        [Required]
+        public string? VersionFile { get; set; }
+
+        [Output]
+        public string? Version { get; set; }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(VersionFile))
+            {
+                Log.LogError($"Could not find version file {VersionFile}");
+                return false;
+            }
+
+            Version = File.ReadLines(VersionFile).First();
```

Comment:
I don't think we need a special task for this. MSBuild has one built in:
https://docs.microsoft.com/en-us/visualstudio/msbuild/readlinesfromfile-task?view=vs-2019

Reply:
yeah you should be able to read the version with something like
```
    <ReadLinesFromFile File="@(VersionFile)">
      <Output TaskParameter="Lines" ItemName="VersionLines" />
    </ReadLinesFromFile>
    <PropertyGroup>
      <EmccVersion>@(VersionLines)</EmccVersion>
    </PropertyGroup>
```

Reply:
That's what is already there. The problem was reading one line, on windows. The whole output is like:
```
emcc (Emscripten gcc/clang-like replacement) 2.0.11 (6e28e4fa4fa1bc50d58b9ddbbb9603a3cf21ea9e)
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

Reply:
Could you use `StartsWith` instead?

```xml
<Error Condition="!$(ActualVersion.StartsWith($(ExpectedVersion))" Text="..." />
```

that solves this issue.

Reply:
Unrelated, how willing would emsdk be to change this command output? It's really weird that asking for a version includes licensing information.

Reply:
We could submit PR

Reply:
The output is probably not designed to be machine parsable. Can't msbuild read the first line only, or extract the first item in @(VersionLines) ?

Reply:
Something like this might work:

```xml
<EmccVersion>@(VersionLines->Reverse())</EmccVersion>
```

Reply:
`<EmccVersion Condition="$([System.String]::new('%(Lines.Identity)').Contains('emcc'))">@(Lines)</EmccVersion>`

Reply:
try `"$([System.IO.File]::ReadAllText('$(VersionsFile)')[0])"`

Reply:
It doesn't let me call `File.ReadAllLines`. So, now I have:
```xml
    <ItemGroup>
      <_ReversedVersionLines Include="@(_VersionLines->Reverse())" />
    </ItemGroup>
    <PropertyGroup>
      <ActualEmccVersion>%(_ReversedVersionLines.Identity)</ActualEmccVersion>
    </PropertyGroup>
```

I think this is a decent enough solution for now, and it doesn't depend on any specific string/char in the output.

Reply:
I meant ReadAllLines but I see that isn't available :(
The Reverse() call looks best